src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -128,7 +129,9 @@ MultiRouterPlanCreate(Query *originalQuery, Query *query,
 	if (modifyTask)
 	{
 		ErrorIfModifyQueryNotSupported(query);
+
 		task = RouterModifyTask(originalQuery, query);
+		requiresMasterEvaluation = RequiresMasterEvaluation(originalQuery);
```

Comment:
Can we set this in RouterQueryJob as it was before, and set it only if task_type = MODIFY_TASK